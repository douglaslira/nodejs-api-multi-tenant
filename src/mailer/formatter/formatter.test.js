'use strict'

import defaultFormatter from './index'

test('formats confirmation email correctly', () => {
  const result = defaultFormatter.format({
    language: 'en',
    type: 'confirm-registration',
    payload: {
      domainName: 'imaginary-domain.com',
      confirmationUrl: `https://imaginary-domain.com/confirm-registration?jwt=abcd.1234.y1z2`,
      user: {
        username: 'joe-user',
        email: 'joe@user.com'
      }
    }
  })
  expect(result.subject).toBe('Confirm Registration')
  expect(result.html).toMatchSnapshot()
})
