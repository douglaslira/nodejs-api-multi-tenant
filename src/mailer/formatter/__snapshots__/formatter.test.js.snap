// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats confirmation email correctly 1`] = `
"<div style=\\"font-family: arial; direction: ltr;\\">
  <h1>Welcome to  imaginary-domain.com</h1>
  <p>Please type this pin into the registration form to complete your registration:</p>
  <p><span class=\\"pin\\" style=\\"font-size: 36px;\\"></span></p>
  <p>Alternatively, please click the link below to complete your registration.</p>
  <a href=\\"https://imaginary-domain.com/confirm-registration?jwt=abcd.1234.y1z2\\">https://imaginary-domain.com/confirm-registration?jwt=abcd.1234.y1z2</a>
</div>
"
`;
